### DISCUSSION TITLE
ROS2 Humble - Topics not publishing on PC

### DISCUSSION THREAD

**Discussion by rohanchandra-cpe** [iRobotEducation/create3_docs#524] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

HP Envy, Ubuntu 22.04.3 LTS (Jammy Jellyfish)

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.3

### Which RMW is your robot running?

I don't know

### Is the robot connected to a network? If so, what is the network type?

Yes: home network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

The create 3 specific topics (e.g. /interface_buttons) are not publishing on my PC, but they are publishing on the raspberry pi. The other topics e.g. /ip, are publishing on both my pc and on the rpi. I don't know whether I am using cyclonedds or fastrtps, but I think I'm using fastrtps (I don't know how to confirm this). Here is an image from me attempting to run a topics echo command on my PC.

**Comment by brianabouchard** [comment]
Hello! Are you using a TurtleBot or just the Create 3 Robot? I ask because the /ip topic isn't one that is used by the Create but it is used by TurtleBot. It only matters in that configurations may be slightly different. 

In the meantime, I would do two things. First, if you're using a RPi wired directly to the Create, you'll want to switch the adapter board to USB so you can leverage the USB0 interface. 

Second, I would check the RMWs of your robot, PC and RPi. Here's how you would do that:

- For the PC and RPi, you should be able to enter the following command into terminal `echo $RMW_IMPLEMENTATION`. That should result in an output of either `rmw_fastrtps_cpp` or `rmw_cyclonedds_cpp`. To change the RMW, see [these instructions](https://iroboteducation.github.io/create3_docs/setup/xml-config/). 
- For the Create 3, you can go to the webserver by navigating in a browser to the IP address of your Create 3 robot. Full details on all you can access on the webserver are available [here](https://iroboteducation.github.io/create3_docs/webserver/overview/), but what you're looking for is the Application page and then a field there labeled RMW_IMPLEMENTATION.

The key here is that the RMW has to be the same for all 3 devices in order for them to communicate properly. I've also been working on a troubleshooting script to help expedite these checks. If it's helpful, you're welcome to test it out on both the Pi and PC and see if it flags any issues: https://github.com/iRobotEducation/create3_docs/blob/brianabouchard/revised-question-template/.github/DISCUSSION_TEMPLATE/troubleshooting.py

Let us know the outcome of these tests!

**Reply by rohanchandra-cpe** [reply]
Hi! Thanks for the reply :)

I looked at all 3 and they are all on rmw_fastrtps_cpp. 
How would I switch my adapter board to USB?
I'm running your script right now (I had to make a venv to pip install requests, and I'm using Python 3.6.9). Here was the result:

![script_run_fail](https://github.com/iRobotEducation/create3_docs/assets/58828165/4436332c-e800-4211-a8b9-6b215f5052f1)

Also, I got the robot ip from the Connect tab on the Create 3 webserver website. Is that the correct one?

**Reply by shamlian** [reply]
The switch is on the [adapter board](https://iroboteducation.github.io/create3_docs/hw/adapter/).

**Reply by ldbrouwer** [reply]
I have a similar problem. I can connect to the Create 3 through the Raspberry Pi and get topics and run scripts, but I am unable to connect to the Create 3  (or at least I can't get the topic list) over wifi. I have  disconnected the RPi and tried to connect with my laptop.  I ran your troubleshooting script and everything came back OK.  I am running H2.6 on  ubuntu 22.04 on both the laptop . I am on my home network (verizon fios router) with only this one robot. I am running fastdds on robot and laptop (Dell Inspiron 15). I also tried to disable multicasting with the XML file on the ROS2 DMS page.  But this yielded no results. Any thoughts? Need more data?
